test/editor/operations-test.js

```diff
@@ -44,6 +45,23 @@ describe('editor/operations: formatting operations', () => {
             expect(model.serializeParts()).toEqual([{ "text": "hello _world_!", "type": "plain" }]);
         });
 
+        it('works for escaping backticks in between texts', () => {
+            const renderer = createRenderer();
+            const pc = createPartCreator();
+            const model = new EditorModel([
+                pc.plain("hello ` world!"),
+            ], pc, renderer);
+
+            const range = model.startRange(model.positionForOffset(0, false),
+                model.positionForOffset(13, false));  // hello ` world
+
+            expect(range.parts[0].text.trim().includes("`")).toBeTruthy();
+            expect(longestBacktickSequence(range.parts[0].text.trim())).toBe(1);
+            expect(model.serializeParts()).toEqual([{ "text": "hello ` world!", "type": "plain" }]);
+            toggleInlineFormat(range, "`".repeat(longestBacktickSequence(range.parts[0].text.trim()) + 1));
+            expect(model.serializeParts()).toEqual([{ "text": "``hello ` world``!", "type": "plain" }]);
+        });
```

Comment:
This test does not call upon the method where your code lives `formatRangeAsCode`